_posts/2017-03-07-scalajs-react-part1.md

```diff
@@ -0,0 +1,316 @@
+---
+layout: post
+author: pedropramos
+title: Building Web applications with Scala.js and React - Part 1
+---
+
+As a Scala programmer developing Web applications, it is usually uncomfortable to move from a tidy, functional, and type-safe Scala back-end to an often subpar JavaScript front-end. Luckily for us, there are already some strong and mature alternatives to the Web's (not so) lovely *lingua franca*.
+
+[**Scala.js**](https://www.scala-js.org/) is an implementation of Scala, authored by [Sébastien Doeraene](https://github.com/sjrd), which compiles Scala code to JavaScript (as opposed to JVM bytecode). It supports full bilateral interoperability between Scala and JavaScript code and therefore allows us to develop front-end Web applications in Scala using JavaScript libraries and frameworks. It also promotes less code duplication for a typical Scala Web application by allowing us to reuse the models developed for the back-end on the front-end.
```

Comment:
when you say reuse the models, I'd say models and common logic (for instance)....